controllers/machinedeploymentupgrade_controller.go

```diff
@@ -206,12 +230,50 @@ func (r *MachineDeploymentUpgradeReconciler) updateStatus(ctx context.Context, l
 	mdUpgrade.Status.Upgraded = int64(nodesUpgradeCompleted)
 	mdUpgrade.Status.RequireUpgrade = int64(nodesUpgradeRequired)
 	mdUpgrade.Status.Ready = nodesUpgradeRequired == 0
+	if mdUpgrade.Status.Ready {
+		machineSpecJSON, err := base64.StdEncoding.DecodeString(mdUpgrade.Spec.MachineSpec)
+		if err != nil {
+			return fmt.Errorf("decoding value for %s with base64: %v", mdUpgrade.Spec.MachineSpec, err)
+		}
+		machineSpec := clusterv1.MachineSpec{}
+		if err := json.Unmarshal(machineSpecJSON, &machineSpec); err != nil {
+			return fmt.Errorf("unmarshalling machineSpec: %v", err)
+		}
+		log.Info("Updating Spec in Machine Set", "machineset", ms.Name)
+		ms.Spec.Template.Spec = machineSpec
+	}
 	return nil
 }
 
+func (r *MachineDeploymentUpgradeReconciler) getCurrentMachineSet(ctx context.Context, md *clusterv1.MachineDeployment) (*clusterv1.MachineSet, error) {
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: map[string]string{mdLabelIdentifier: md.Name}})
+	if err != nil {
+		return nil, err
+	}
+	machineSetsList := &clusterv1.MachineSetList{}
+	if err := r.client.List(ctx, machineSetsList, &client.ListOptions{LabelSelector: selector}); err != nil {
+		return nil, fmt.Errorf("getting machine sets for %v: %v", md.Name, err)
+	}
+	revision, ok := md.Annotations[clusterv1.RevisionAnnotation]
```

Comment:
Does field selector work for annotations or is that specifically for fields under spec? That might help avoid the for loop below right?